src/snarky.d.ts

```diff
@@ -51,489 +53,767 @@ declare interface ProvablePure<T> extends Provable<T> {
+   * 
+   * @return An array of {@link Bool} element representing little endian binary representation of this {@link Field}.
    */
   toBits(length: number): Bool[];
 
   /**
-   * Check if this {@link Field} equals another {@link Field}-like value.
-   * Returns a {@link Bool}.
+   * Check if this {@link Field} is equal another `field-like` value.
+   * Returns a {@link Bool}, which is a `provable` type and can be used to prove the validity of this statement.
    *
    * ```ts
-   * Field(2).equals(2); // Bool(true)
+   * Field(5).equals(5).assertEquals(Bool(true));
    * ```
+   * 
+   * @param value - the `field-like` value to compare with this {@link Field}.
+   * 
+   * @return A {@link Bool} representing if this {@link Field} is equal another `field-like` value.
    */
-  equals(y: Field | number | string | boolean): Bool;
+  equals(value: Field | number | string | boolean): Bool;
 
-  // TODO: Izzy to document
+  /**
+   * **Warning: This function is mainly for internal use. Normally it is not intended to be used by a zkApp developer.
+   * 
+   * In snarkyJS, addition and scaling (multiplication of variables by a constant) of variables is represented as an AST - [abstract syntax tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree). For example, the expression `x.add(y).mul(2)` is represented as `Scale(2, Add(x, y))`.
+   * 
+   * Only when the variable is needed in a multiplicative or any higher level constraint (for example multiplication of two {@link Field} elements) a new internal variable is created to represent the operation.
+   * 
+   * The `seal()` function tells snarkyJS to stop building an AST and create a new variable right away.
+   * 
+   * @return A {@link Field} element that is equal to the result of AST that was previously on this {@link Field} element.
+   */
   seal(): Field;
-  // TODO: Izzy to document
+  
+  /**
+   * Create a new {@link Field} element from the first `numBits` of this {@link Field} element.
+   * As {@link Field} elements are represented using [little endian binary representation](https://betterexplained.com/articles/understanding-big-and-little-endian-byte-order/), the resulting {@link Field} element will equal the original one if the variable fits in `numBits` bits.
+   * 
+   * @param numBits - The number of bits to take from this {@link Field} element.
+   * 
+   * @return A {@link Field} element that is equal to the `numBits` of this {@link Field} element.
+   */
   rangeCheckHelper(numBits: number): Field;
 
   /**
-   * Checks whether this is a hard-coded constant in the Circuit.
+   * Check whether this {@link Field} element is a hard-coded constant in the Circuit.
+   * If a {@link Field} is constructed outside a zkApp method, it is a constant.
+   * 
+   * ```ts
+   * console.log(Field(42).isConstant()); // False
+   * ```
+   * 
+   * ```ts
+   * @method(x: Field) {
+   *    console.log(x.isConstant()); // True
```

Comment:
```suggestion
   *    console.log(x.isConstant()); // False
```
you mixed them up :D